onnx/onnx.in.proto

```diff
@@ -430,8 +487,22 @@ message ModelProto {
   // One FunctionProto can reference other FunctionProto in the model, however, recursive reference
   // is not allowed.
   repeated FunctionProto functions = 25;
+
+  // Describes different target configurations for a multi-device use case.
+  // A model can describe multiple multi-device configurations for execution.
+  repeated DeviceConfigurationProto configuration = 26;
 };
 
+// DeviceConfigurationProto describes a multi-device configuration for a model.
+message DeviceConfigurationProto {
+    // Name of the configuration.
+    optional string name = 1;
+    // Name of the device.
```

Comment:
I think the following may not cover pipelining between different kinds of devices (for example, between 1 cpu and 1 gpu). Should we replace the following by a simple `repeated string device_name`? So, it could be ["cpu", "gpu"] or ["gpu1", "gpu2", "gpu3", "gpu4"]. Of course, more specific names can also be used, like "com.nvidia:H100" etc. We may need some naming conventions.